pimd/pim_mroute.c

```diff
@@ -97,7 +97,7 @@ static int pim_mroute_set(struct pim_instance *pim, int enable)
 	if (enable) {
 		/* Linux and Solaris IP_PKTINFO */
 		data = 1;
-		if (setsockopt(pim->mroute_socket, IPPROTO_IP, IP_PKTINFO,
+		if (setsockopt(pim->mroute_socket, PIM_IPPROTO, IP_PKTINFO,
```

Comment:
This doesn't work, `IP_PKTINFO` is not valid for `IPPROTO_IPV6`

Reply:
pim_mroute_set is defined for IPV4 and IPV6 separately .Line no 693 has pim_mroute_set for V6

Reply:
Ah right. Well — then there is no point in using `PIM_IPPROTO` here…